app/helpers/samples_helper.rb

```diff
@@ -367,6 +420,18 @@ def format_samples(samples)
     formatted_samples
   end
 
+  def visibility(samples)
+    # When in conjunction with some filters, the query below was not returning the public property,
+    # thus we need to get ids and redo the query independently
+    sample_ids = samples.pluck(:id)
+    return current_power
+           .samples
+           .where(id: sample_ids)
+           .joins(:project)
+           .select("samples.*", "IF(projects.public_access = 1 OR DATE_ADD(samples.created_at, INTERVAL projects.days_to_keep_sample_private DAY) < '#{Time.current.strftime('%y-%m-%d')}', true, false) AS public")
```

Comment:
hmmm... this is getting complicated... it seems like long-run we should simplify, centralize and-or write test code so that we don't ever make mistakes with visibility and it doesn't cost in perf

Reply:
Testing question is a trade-off between speed of development and correctness.
I agree and we discussed before that our bar is currently low and we need more tests for correctness. Just need time/resources.

This is not a critical path since it works on a set of samples and access control should be done through current_power/consul.